assignment-3.md

```diff
@@ -0,0 +1,108 @@
+# Assignment 3: Tests
+
+## 3.1 Getting Started
+
+Start the test runner with `npm test`.
+
+By default only tests for changed files are run. So in case you have been making
+Git commits, and currently have no uncommitted changes in your working
+directory, press `a` to trigger a test run.
+
+The initial test in `Climate.test.tsx` probably does not run green anymore,
+because it was testing the app in its starting state, but we have changed the
+app quite a bit in the last exercises.
+
+We'll take care for that soon, but for now disable that test with Jest's `skip`
+feature. Change this:
+
+```typescript
+test('Climate app shows the current temperature…', …)
+```
+
+to this:
+
+```typescript
+test.skip('Climate app shows the current temperature…', …)
+```
+
+## 3.2 Header
+
+Let's get started by writing some tests for the `Header` component that we
+have implemented in the first assignment. You have probably put that into a
+file called `Header.tsx`. Create a new file `Header.test.tsx` for the tests.
+
+Try to `render` the component with different text values that you pass in as a
+prop. Use `getByRole` and `toHaveTextContent` to find the rendered header in the
+DOM and check its text content.
+
+*Optional:* If you're feeling fancy, you can use
+[Jest's `test.each`](https://jestjs.io/docs/en/api#testeachtablename-fn-timeout)
+to create a parametrized test.
```

Comment:
I'd drop that paragraph, probably only confuses the participants

Reply:
ACK

Reply:
Done.